plugins/utils/persistence.nix

```diff
@@ -0,0 +1,68 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib; let
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options.plugins.persistence =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "persistence.nvim";
+
+      package = helpers.mkPackageOption "persistence.nvim" pkgs.vimPlugins.persistence-nvim;
+
+      dir = helpers.defaultNullOpts.mkNullable (with types; either str helpers.rawType) ''
+        vim.fn.expand(vim.fn.stdpath("state") .. "/sessions/")
+      '' "directory where session files are saved";
```

Comment:
```suggestion
      dir =
        helpers.defaultNullOpts.mkNullable
        (with types;
          either str helpers.rawType
        )
        "vim.fn.expand(vim.fn.stdpath("state") .. "/sessions/")"
        "directory where session files are saved";
```

Reply:
It would not work because there are quotes (`"`) in the string. I wanted to avoid escaping them so I used a `'' ''` string